frame/scheduler/src/lib.rs

```diff
@@ -144,8 +178,8 @@ decl_module! {
 			priority: schedule::Priority,
 			call: Box<<T as Trait>::Call>,
 		) {
-			ensure_root(origin)?;
-			let _ = Self::do_schedule(when, maybe_periodic, priority, *call);
+			T::ScheduleOrigin::ensure_origin(origin.clone())?;
+			let _ = Self::do_schedule(when, maybe_periodic, priority, origin.caller().clone(), *call);
```

Comment:
is it possible to avoid this clone?

Reply:
you can probably fix the `Success` type of `T::ScheduleOrigin` and use the result from `ensure_origin(..)?`

Reply:
The `Success` type of `EnsureRoot` is `()` so that will need quite a few changes.